sequence_processing_pipeline/ConvertJob.py

```diff
@@ -210,3 +218,158 @@ def parse_job_script(job_script_path):
+        qiita_ids = bioinformatics['QiitaID'].tolist()
+
+        if 'contains_replicates' in sheet.Bioinformatics:
+            has_reps = sheet.Bioinformatics['contains_replicates'].tolist()
+            # assume a validated sample-sheet ensures has_reps has only one
+            # value, either True or False.
+            self.contains_replicates = bool(has_reps[0])
+        else:
+            self.contains_replicates = False
+
+        results = {}
+
+        for project, qiita_id in zip(projects, qiita_ids):
+            # derive project_name by removing the known qiita_id associated
+            # with this project from the string.
+            project_name = re.sub(f'_{qiita_id}$', '', project)
+            results[project] = {'qiita_id': qiita_id,
+                                'project_name': project_name,
+                                'full_project_name': project,
+                                'samples': {}}
+
+        # Since the project-name is stored in an internal variable
+        # in a third-party library, convert samples to JSON using the exposed
+        # method first.
+        samples = json_loads(sheet.to_json())['Data']
+
+        for sample in samples:
+            d = {'Sample_Name': sample['Sample_Name'],
+                 'Sample_ID': sample['Sample_ID'],
+                 # matching files will store the paths to all fastq files
+                 # associated w/this sample-name.
+                 'matching_files': []}
+
+            if 'orig_name' in sample:
+                d['orig_name'] = sample['orig_name']
+
+            results[sample['Sample_Project']]['samples'][d['Sample_Name']] = d
+
+        # associate with each dictionary a list of matching fastq files.
+        # this way, we can easily determine which files to copy to another
+        # project based on just a sample-name/sample-id and a project-name.
+
+        for project in results:
+            # find just the fastq files for this project.
+
+            fastq_paths = self._find_files(join(self.output_path,
+                                                project))
+            fastq_paths = [f for f in fastq_paths if f.endswith('.fastq.gz')]
+
+            for sample_name in results[project]['samples']:
+                sample = results[project]['samples'][sample_name]
+                # regex based on studying all filenames of all fastq files in
+                # $WKDIR. Works with _R1_, _R2_, _I1_, _I2_, etc.
+                rgx = r"^" + re.escape(sample['Sample_ID']) + \
+                      r"_S\d+_L\d+_[R,I]\d+_\d+.fastq.gz$"
+
+                for full_path in fastq_paths:
+                    file_path, file_name = split(full_path)
+                    if re.match(rgx, file_name):
+                        sample['matching_files'].append(full_path)
+
+        self.info = results
+
+    def copy_sequences(self, sample_name, source_project, dest_project,
+                       copy_all_replicates=False):
```

Comment:
should there also be a flag for only "tech" samples (blanks, katharo, etc)?

Reply:
@antgonza My understanding is that copying is done at the *per sample name* level (either the real sample name or in the case of replicates, the orig name). I had expected that something outside this method (which @charles-cowart and I have discussed but haven't implemented yet) would identify all the tech samples to be transferred, and would then call this method once for each of them.  In this model, how would a "tech_samples_only" flag on this method be used?

Reply:
Thank you for the explanation; I get it now: this is a general _per sample_name_ copy method and there is going to be another PR where it's decided what to copy.